onnxruntime/contrib_ops/cuda/bert/attention_strided_copy.cu

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cuda_common.h"
+#include "contrib_ops/cuda/bert/attention_impl.h"
+
+using namespace onnxruntime::cuda;
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+template <typename T>
+__global__ void StridedCopy(const T* in, const int H, longlong4 in_strides,  // coord (b,n,s,h)
+                            T* out, longlong4 out_strides                    // coord (b,n,s,h)
+) {
+  const int h = threadIdx.x;
+  const int n = threadIdx.y;
+  const int s = blockIdx.x;
+  const int b = blockIdx.y;
+  if (h < H) {
+    const int in_offset = b * in_strides.x + n * in_strides.y + s * in_strides.z + h * in_strides.w;
+    const int out_offset = b * out_strides.x + n * out_strides.y + s * out_strides.z + h * out_strides.w;
+    out[out_offset] = in[in_offset];
+  }
+}
+
+template <typename T>
+__global__ void StridedCopyLarge(const T* in, const int H, longlong4 in_strides,  // coord (b,n,s,h)
+                                 T* out, longlong4 out_strides                    // coord (b,n,s,h)
+) {
+  // Use when (H*)*num_heads > 1024
+  int h = threadIdx.x;
+  const int n = threadIdx.y;
+  const int s = blockIdx.x;
+  const int b = blockIdx.y;
+
+  const int h_step = blockDim.x;
+
+  while (h < H) {
+    const int in_offset = b * in_strides.x + n * in_strides.y + s * in_strides.z + h * in_strides.w;
+    const int out_offset = b * out_strides.x + n * out_strides.y + s * out_strides.z + h * out_strides.w;
+    out[out_offset] = in[in_offset];
+    h += h_step;
+  }
+}
+
+template <int NumBytes>
+struct ToByteType;
+
+template <>
+struct ToByteType<2> {
+  using T = uchar2;
```

Comment:
Is it better to use fixed length type T=int16_t, int32_t, int64_t for 2/4/8?

Reply:
I am thinking about switching to use half2/4 and float2/4. Because if I want to add qkv bias support, treating them as bytes is not a proper way.

I think I will do it in later PR, tho.

Reply:
Adopted